pyccel/stdlib/STC_Extensions/Set_extensions.h

```diff
@@ -56,5 +56,34 @@ static inline void _c_MEMB(_intersection_update)(Self* self, Self* other) {
         }
     }
 }
+#if defined (i_use_cmp) 
+// Function to get the minimum element from the set
+static inline i_key _c_MEMB(_min)(const Self* self) {
+    _c_MEMB(_iter) itr = _c_MEMB(_begin)(self);
+    i_key min_val = *(itr.ref);
+    while (itr.ref)
+    {
+        if (*(itr.ref) < min_val)
```

Comment:
Shouldn't you use i_less to ensure that this keeps working if we add complex support later?

Reply:
fixed!